classes/client/s3_client.php

```diff
@@ -122,7 +122,7 @@ private function get_md5_from_hash($contenthash) {
     public function verify_object($contenthash, $localpath) {
         $localmd5 = md5_file($localpath);
         $externalmd5 = $this->get_md5_from_hash($contenthash);
-        if ($externalmd5) {
+        if ($externalmd5 === $localmd5) {
```

Comment:
So this is what we used to do, but the code was changed to just check that we could get ant kind of externalmd5 when we discovered the undocumented change that the etag that AWS stores is not always the md5. So this check essentially became an is readable check. Making this change will actually break this plugin for a subset of objects.

Reply:
See https://forums.aws.amazon.com/thread.jspa?messageID=203436&#203436

Reply:
@kenneth-hendricks is this true of all stores? And was this introduced when we implemented multi part upload, or on the aws side?

@golenkovm whatever the outcome, bake this into a comment please

Reply:
@brendanheywood verify_object method changed for all storages, the note is added